### PR TITLE
Modernize the addon

### PR DESCRIPTION
**Pull request by alexlafroscia**
- Install Yarn
- Update addon for `ember-cli` 2.17.1
- Rewrite tests using async/await and the latest `ember-qunit` API
- Clean up some of the component code to avoid now-deprecated APIs
- Avoid accessing the Ember global object

### CODE REVIEW

**Comment by alexlafroscia** [comment]
I know that you're looking for someone to pick up the maintainence. If you're interested in giving me direct commit access and allowing me to publish to NPM, I'm happy to help out.

**Comment by alexlafroscia** [comment]
Closes #24, which is about installing the Bower dependencies automatically. Bower is no longer a requirement after this PR.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  root: true,
+  parserOptions: {
+    ecmaVersion: 2017,
+    sourceType: 'module'
+  },
+  extends: ['eslint:recommended', 'prettier'],
+  plugins: ['prettier'],
```
Comment (wesm87) [inline .eslintrc.js]: 👍 for prettier

**Comment by alexlafroscia** [comment]
Thanks, I agree on both points. I also want to take a pass at moving the rest of the unit tests for the component and turning them into integration tests, which are more "correct" for testing a component anyway. Will put a follow-up PR together.

**Pull request merged**